source/common/config/api_type_oracle.cc

```diff
@@ -31,5 +31,27 @@ ApiTypeOracle::getEarlierVersionMessageTypeName(const std::string& message_type)
   }
   return absl::nullopt;
 }
+
+const absl::string_view ApiTypeOracle::typeUrlToDescriptorFullName(absl::string_view type_url) {
+  const size_t pos = type_url.rfind('/');
+  if (pos != absl::string_view::npos) {
+    type_url = type_url.substr(pos + 1);
+  }
+  return type_url;
+}
+
+const std::string ApiTypeOracle::descriptorFullNameToTypeUrl(std::string& type) {
```

Comment:
The utilities are more general than just the API type oracle, hence I think they should be in their own standalone library in `source/common/protobuf`.

Reply:
Yes, that sounds reasonable.